docs/src/dev-docs/architecture-life-cycle.rst

```diff
@@ -0,0 +1,56 @@
+Life Cycle of an Architecture
+=============================
+
+.. TODO: Maybe add a flowchart later
+
+Architectures in `metatensor-models` undergo different stages based on their
+development/functionality level and maintenance status. We distinguish three distinct
+stages: **experimental**, **stable**, and **deprecated**. Typically, an architecture
+starts as experimental, advances to stable, and eventually becomes deprecated before
+removal if maintenance is no longer feasible.
+
+.. note::
+    The development and maintenance of an architecture must be fully undertaken by the
+    architecture's authors or maintainers. The core developers of `metatensor-models`
+    provide infrastructure and implementation support but are not responsible for the
+    architecture's internal functionality or any issues that may arise therein.
+
+Experimental Architectures
+---------------------------
+
+New architectures added to the library will initially be classified as experimental.
+These architectures are stored in the ``experimental`` subdirectory within the
+repository. To qualify as an experimental architecture, certain criteria must be met:
+
+1. Capability to fit at least a single quantity and predict it, verified through CI
+   tests.
+2. Compatibility with JIT compilation using `torch script
+   <https://pytorch.org/docs/stable/jit.html>`_.
+3. Provision of reasonable default hyperparameters.
+4. Minimal code quality, ensured by passing linting tests invoked with ``tox -e lint``.
+5. A responsive contact person designated as the maintainer.
+6. All external dependencies must be pip-installable. While not required to be on PyPI,
+   a public git repository or another public URL with a repository is acceptable.
+
+For detailed instructions on adding a new architecture, refer to
+:ref:`adding-new-models`.
+
+Stable Architectures
+--------------------
+
+Transitioning from an experimental to a stable model requires additional criteria to be
+satisfied:
+
+1. Provision of regression prediction tests with a small (not exported) checkpoint file.
+2. Comprehensive documentation explaining the model's hyperparameters.
```

Comment:
I don't fully agree with this. I would say that the hypers can be explained quickly, but what is needed is a guide on how to tune the hypers for different use cases

Reply:
I changed this to `2. Comprehensive architecture documentation`. This is better.

Reply:
Perfect.